lib/network/modules/components/nodes/util/CircleImageBase.js

```diff
@@ -14,12 +14,22 @@ class CircleImageBase extends NodeBase {
   }
 
   setImages(imageObj, imageObjAlt) {
-    if (imageObj) {
-      this.imageObj = imageObj;
+    if (!this.selected) {
+      if (imageObj) {
+        this.imageObj = imageObj;
 
-      if (imageObjAlt) {
-        this.imageObjAlt = imageObjAlt;
+        if (imageObjAlt) {
```

Comment:
OK fine.

> I'm too used to TypeScript now :)

Oh, you lucky bastard. I'm itching to go there. There is no hope for TypeScript in `vis.js`, but I've been rabidly refactoring my own current project to make it possible.

Thank you sincerely for your effort here. The bug is genuine and it needs to be tackled. All the best!